src/cluster/session.rs

```diff
@@ -230,6 +224,43 @@ where
     Ok(session)
 }
 
+/// This function uses a user-supplied connection configuration to initialize all the
+/// connections in the session. It can be used to supply your own transport and load
+/// balancing mechanisms in order to support unusual node discovery mechanisms
+/// or configuration needs.
+///
+/// The config object supplied differs from the ClusterTcpConfig and ClusterRustlsConfig
+/// objects in that it is not expected to include an address. Instead the same configuration
+/// will be applied to all connections across the cluster.
+pub async fn connect<T, M, C, LB>(
```

Comment:
It would be good to append `_static` to this fn and create an equivalent for unstable-dynamic-cluster feature. Also, can you add an example to the examples folder on how to use it

Reply:
Sure. It might take me a few days to get that in, but I'll see what I can do along those lines. Would it be possible to promise the 'example' implementation as a separate follow-up PR? That would probably let me turnaround on the other parts more quickly.

Reply:
Sure, we can add an example later.

Reply:
One problem with suffixing _static is that there's already a connect_static so maybe this needs a more descriptive name? `user_connect[_static|_dynamic]` maybe?